design-proposals/feature-lifecycle.md

```diff
@@ -0,0 +1,327 @@
+- As a KubeVirt cluster operator, I would like to evaluate an
+  undergraduate ("Beta") feature in a real-life environment with actual users.
+- As a KubeVirt cluster operator, I would like to keep using a feature
+  that got graduated after I used it during the "Beta" evaluation period.
+- As a KubeVirt cluster operator, I would like to know that
+  an experimental ("Alpha") feature is planned to be removed.
+- As a KubeVirt cluster operator, I would like to know that
+  an undergraduate ("Beta") feature is planned to be removed.
+- As a KubeVirt cluster operator, I would like to stop using a feature
+  that got removed.
+
+### Repos
+This is a cross repo project policy under the
+[kubevirt](https://github.com/kubevirt) organization.
+
+## Proposal Design
+The proposal on how to define a feature lifecycle is influenced by
+processes and policies from the Kubernetes project.
+These sources are scattered around, each focusing on different
+aspects of a feature:
+- [Feature Gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
+- [Graduation](https://kubernetes.io/blog/2020/08/21/moving-forward-from-beta/)
+- [Changing the API](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md)
+- [Deprecation](https://kubernetes.io/docs/reference/using-api/deprecation-policy/)
+
+The proposal takes the top-down approach, starting with the high level
+flow that a common feature will traverse through.
+Continuing with actions that need to be taken and timeline suggestions.
+
+Both feature graduation and discontinuation flows are covered.
+Including the implications on users.
+
+Depending on individual topics, follow-up proposals may extend the basic
+points raised in this proposal.
+
+### Terminology
+Feature Gates and feature configuration are often used interchangeable or not differentiated.
+- Feature Gate: A flag that controls the presence or availability of a feature in the cluster.
+- Feature configuration: Cluster or workload level configuration that allows an admin or user
+  (depending on the feature) to control aspects of a feature operation.
+  A common usage is to determine if features are opt-in or opt-out by default.
+
+### Feature Stages
+A feature is expected to pass in the following order through the following stages:
+1. Enhancement proposal.
+2. Implementation.
+3. Release as Alpha (experimental).
+4. Release as Beta (pre-release for evaluation).
+5. Release as General Availability (graduation).
+6. Removal.
+
+Starting from the Alpha release, it can be removed with restrictions that
+depend on the release stage (Alpha, Beta, GA).
+
+[Removal](#removal) of features is widely discussed later
+in this proposal.
+
+#### Enhancement proposal
+As the first step for introducing a new feature, a formal proposal is
+expected to be shared for public review via mailinglist and
+a [design proposal](https://github.com/kubevirt/community/tree/main/design-proposals).
```

Comment:
Maybe we should also add  that everyone doesn't have to agree on the mailing list, just that the announcement needs to be made.

Reply:
I prefer this to be open and simple. How the feedback should be processed is partially cultural, partially related to the content.

If we enter into all these little details, this will become a kind of legal document, it is not.
It is aimed to describe a process of managing features in the spirit of how the organization is behaving in general.

Reply:
Are we planning on enforcing this on anyone after it gets merged?